### PR TITLE
Круглов - Лабораторная работа №2

### PR DESCRIPTION
**Pull request by EOKruglov**

### CODE REVIEW

**Comment by EOKruglov** [comment]
@alexsoll @likholat  review pls

File: modules/string-calculator/include/string_calculator.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 Kruglov Yegor
+
+#pragma once
+#include <stack>
+#include <string>
```
Comment (alexsoll) [inline modules/string-calculator/include/string_calculator.h]: pragma once directive is not needed. Error multiple inclusion of header files fixes design ifndef - endif

File: modules/string-calculator/include/string_calculator.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#ifndef MODULES_STRING_CALCULATOR_INCLUDE_STRING_CALCULATOR_H_
+#define MODULES_STRING_CALCULATOR_INCLUDE_STRING_CALCULATOR_H_
+
+class StringCalculator{
+ private:
+    std::string infix, postfix;
+    std::stack<char> charStack;
+    std::stack<double> doubleStack;
```
Comment (alexsoll) [inline modules/string-calculator/include/string_calculator.h]: The names of private variables usually end in a symbol _

File: modules/string-calculator/include/string_calculator.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    void set_infix(const std::string str);
+    void set_postfix(const std::string str);
+    void fill_stacks(int n);
+    std::string get_infix();
+    std::string get_postfix();
+
+    bool check();
+    int priority(const char& el);
+    void to_postfix();
+    double calculate();
```
Comment (alexsoll) [inline modules/string-calculator/include/string_calculator.h]: I think that it is not necessary to give the user the ability to access some internal functions. It only needs to calculate

File: modules/string-calculator/include/string_calculator.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    std::string infix_, postfix_;
+    std::stack<char> charStack_;
+    std::stack<double> doubleStack_;
+
+    bool check();
+    int priority(const char& el);
+    void to_postfix();
+
+ public:
+    void set_infix(const std::string str);
```
Comment (alexbatashev) [inline modules/string-calculator/include/string_calculator.h]: https://google.github.io/styleguide/cppguide.html#Function_Names

**Pull request merged**